src/macros.jl

```diff
@@ -278,6 +265,49 @@ function assert_validmodel(m, macrocode)
     end
 end
 
+"""
+    macro_return(model, code, variable)
+
+Return a code that runs `code` in a local scope which returns the value of `variable`.
+"""
+function macro_return(code, variable)
+    return quote
+        let
+            # The let block ensures that all variables created behaves like
+            # local variables, see https://github.com/JuliaOpt/JuMP.jl/issues/1496
+            # To make $variable accessible from outside we need to return it at
+            # the end of the block
+            $code
+            $variable
+        end
+    end
+end
+
+"""
+    macro_assign_and_return(model, code, variable, registerfun::Function,
+                            quotvarname, escvarname, final_variable=variable)
+
+Return runs `code` in a local scope which returns the value of `variable`
+and then assign `final_variable` to `escvarname`.
+If `registerfun` is given, it registers the value returned by the local scope
```

Comment:
`registerfun` is a function passed in. How do we know that it "registers the value returned by the local scope"?
The docstring should either provide the expected signature or call a concrete function.